app/services/docx_listing_splitter.rb

```diff
@@ -0,0 +1,86 @@
+require "ole/storage"
+
+class DocxListingSplitter
+  def self.split(id, file, &)
+    new(id, file).split(&)
+  end
+
+  def initialize(id, file)
+    @id = id
+    @file = file
+  end
+
+  def split(&block)
+    zip_file = create_zip_version_of_source_file
+    file_names = unzip_attachments_and_list_file_names(zip_file)
+
+    block.call(file_names)
+  ensure
+    delete_extracted_files(zip_file)
+  end
+
+  private
+
+  def create_zip_version_of_source_file
```

Comment:
Should this method name be renamed (and the `zip_file` variable name), since we're not creating a zip version anymore? Though we are still using the `Zip::File` to unzip it, I feel like it is a little confusing since there is no actual Zip file creation happening.

Reply:
Right, interesting point. Some thoughts:

- we are definitely writing out a Zip file, but its filename ends in `.docx`, so calling the variable `zip_file` could make sense.
- we aren't creating a Zip version of a SourceFile, so the method name doesn't make sense.

I'm going to rename the method to `download_file`.

Reply:
Sounds good. I think leaving the file name as `zip_file` works and won't be confusing as we are unzipping it in the next method.